pkg/util/instrument/instrument.go

```diff
@@ -0,0 +1,24 @@
+package instrument
+
+import (
+	"context"
+	"time"
+
+	"github.com/weaveworks/common/instrument"
+)
+
+// ObserveRequest reports how much time was spent on the given function  `f`.
+//
+// It is a thinner version of weaveworks/common/instrument.CollectedRequest that doesn't emit spans.
+func ObserveRequest(ctx context.Context, method string, col instrument.Collector, toStatusCode func(error) string, f func(context.Context) error) error {
```

Comment:
Maybe you can all this function `TimeRequest` to make it clearer what it is observing.

Reply:
that's fair. naming is too hard :laughing: